consensus/event_loop.go

```diff
@@ -21,18 +22,19 @@ func NewFastHotStuffEventLoop() *FastHotStuffEventLoop {
 // params:
 //   - blockConstructionInterval: block construction duration must be > 0
 //   - timeoutBaseDuration: timeout base duration must be > 0
-//   - chainTip: the input block must have a valid block hash, block height, view, and QC
-//   - validators: the validators must be sorted in decreasing order of stake, with a
-//     consistent tie breaking scheme. The validator set is expected to be valid for
-//     validating votes and timeouts for the next block height.
+//   - tip: the current tip of the blockchain, with the validator set at that block height
```

Comment:
Ah good question. It's whatever the tip is in blockchain.go. It'll almost always be the uncommitted tip. But it could be the committed tip in the edge case during the PoW -> PoS cutover. Either way, this code is agnostic to the status of the block. I've added a comment to say the block may be either committed or uncommitted.